File: 0277-find-the-celebrity/0277-find-the-celebrity.cpp
/* The knows API is defined for you.
      bool knows(int a, int b); */

class Solution {
public:
    int findCelebrity(int n) {
        vector<pair<int,int>> knowMap(n); 

        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                if(i == j) continue; 

                if(knows(i,j)){
                    knowMap[i].first++; 
                    knowMap[j].second++; 
                }
            }
        }

        for(int i = 0; i < knowMap.size(); i++){
            if(knowMap[i].first == 0 && knowMap[i].second == n - 1) return i; 
        }
        return -1; 
    }
};

File: 0277-find-the-celebrity/0277-find-the-celebrity.java
/* The knows API is defined in the parent class Relation.
      boolean knows(int a, int b); */

public class Solution extends Relation {
    public int findCelebrity(int n) {
        int[] numKnows = new int[n]; 
        int[] numKnowsWho = new  int[n];

        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                if(i == j) continue; 
                if(knows(i,j)){
                    numKnows[j]++; 
                    numKnowsWho[i]++; 
                }
            }
        }

        for(int i = 0; i < n; i++){
            if(numKnows[i] == n-1 && numKnowsWho[i] == 0) return i; 
        }
        return -1; 
    }
}